### PR TITLE
[repo] Refactor release creation into a powershell module

### PR DESCRIPTION
**Pull request by CodeBlanch**
## Changes

* Refactors the powershell to create GitHub release into a module
* Updates the script to the main repo version. It will now add CHANGELOG content into the release description for the tag being released

### CODE REVIEW

File: build/scripts/post-release.psm1 [hunk 1]
```diff
@@ -0,0 +1,95 @@
+$gitHubBotUserName="github-actions[bot]"
+$gitHubBotEmail="41898282+github-actions[bot]@users.noreply.github.com"
```
Comment (Kielek) [inline build/scripts/post-release.psm1:2]: Just for curiosity - what is the source of the email? Is it documented somehow?

**Review by Kielek — APPROVED** [review]
I'am not pwsh master, but LGTM.

It will be great to see results of the script execution on some fork.

File: build/scripts/post-release.psm1 (hunk 1 above)
Comment (CodeBlanch) [reply build/scripts/post-release.psm1:2]: That is the GitHub "noreply" email address format which is actually supported for all users. Wish I would have known that before 🤣

As far as the ID: https://[api.github.com/users/github-actions%5Bbot%5D](https://api.github.com/users/github-actions%5Bbot%5D)

A discussion: https://github.com/orgs/community/discussions/26560

Comment (Kielek) [reply build/scripts/post-release.psm1:2]: Thanks!

Comment (CodeBlanch) [reply build/scripts/post-release.psm1:2]: Here is a test run: https://github.com/CodeBlanch/opentelemetry-dotnet-contrib/actions/runs/9099109966/job/25010904161

Here is the release it created: https://github.com/CodeBlanch/opentelemetry-dotnet-contrib/releases/tag/Exporter.OneCollector-1.9.0

**Review by vishweshbankwar — APPROVED** [review]

File: build/Projects/OpenTelemetry.Instrumentation.AspNet.proj [hunk 2]
```diff
@@ -14,7 +14,8 @@
     <PackProjects Include="$(RepoRoot)\src\OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule\OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule.csproj" />
     <PackProjects Include="$(RepoRoot)\src\OpenTelemetry.Instrumentation.AspNet\OpenTelemetry.Instrumentation.AspNet.csproj" />
 
-    <TestProjects Include="$(RepoRoot)\test\OpenTelemetry.Instrumentation.AspNet*.Tests\*.Tests.csproj" />
```
Comment (CodeBlanch) [inline build/Projects/OpenTelemetry.Instrumentation.AspNet.proj:17]: FYI the wildcard here `OpenTelemetry.Instrumentation.AspNet*` was causing `AspNetCore` tests to be included which weren't part of the build so test run were failing for AspNet proj. Took me a while to figure that out 🤣

File: build/scripts/post-release.psm1 (hunk 1 above)
Comment (CodeBlanch) [reply build/scripts/post-release.psm1:2]: Here is a test run for AspNet stuff: https://github.com/CodeBlanch/opentelemetry-dotnet-contrib/actions/runs/9099945737/job/25013678335

Here is the release it created: https://github.com/CodeBlanch/opentelemetry-dotnet-contrib/releases/tag/Instrumentation.AspNet-1.9.0-alpha.1

**Review by rajkumar-rangaraj — APPROVED** [review]

**Pull request merged**